### PR TITLE
WIP. Document how to create a plugin

### PR DESCRIPTION
**Pull request by cmoulliard**
- Document how to create a plugin and steps needed
- How to build and test it

### CODE REVIEW

**Comment by cmoulliard** [comment]
@metacosm Question: Where should we explain the process to release a plugin and next to use it with halkyon operator ?

File: README.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+2020-02-13T16:24:06.411+0100 [DEBUG] example-plugin.example-plugin: 2020-02-13T16:24:06.411+0100 [INFO ] example-plugin.PluginResource: calling GetSupportedTypes
+2020-02-13T16:24:06.411+0100 [DEBUG] example-plugin.example-plugin: 2020-02-13T16:24:06.411+0100 [DEBUG] example-plugin: plugin address: network=unix address=/var/folders/56/dtp67r4n1hv79q2hrh_dbwcc0000gn/T/plugin731069597
+2020-02-13T16:24:06.412+0100 [DEBUG] example-plugin: using plugin: version=1
+```
+
+## Debug
+
+- If you plan to debug the `rpc` plugin, simply add the following compilation parameters to the `go build` command
+  ```bash
+  go build -gcflags="all=-N -l" -o example-plugin ./cmd/example-capability/plugin.go
```
Comment (metacosm) [inline README.md:78]: Is this really needed?

File: README.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
+# Instructions to create a new plugin
+
+The process to create a new `plugin` able to manage a new capability for the `halkyon` operator is pretty forward
```
Comment (metacosm) [inline README.md]: forward -> straightforward

File: README.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
+# Instructions to create a new plugin
+
+The process to create a new `plugin` able to manage a new capability for the `halkyon` operator is pretty forward 
+and consists in a few steps :
+
+- Git clone the project `https://github.com/halkyonio/plugin-example-capability` and rename it 
+- Rename the `module name` as defined within the `go.mod` file to use your package name
+- Find and replace the `example` word with the name of your `plugin`, resource, ...
```
Comment (metacosm) [inline README.md]: That's not really detailed enough, not very clear…

File: README.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+## Debug
+
+- If you plan to debug the `rpc` plugin, simply add the following compilation parameters to the `go build` command
+  ```bash
+  go build -gcflags="all=-N -l" -o example-plugin ./cmd/example-capability/plugin.go
+  ```
+
+- Next, start the Operator in Debug mode. During the launch of the operator, the plugin will be loaded and the RPC server started.
+- Use the `Attach to Process` debug option of your IDE to start within this project a debugger.
```
Comment (metacosm) [inline README.md:82]: This instruction is IDEA-specific and I'm not sure how to perform this step without IDEA, actually… :(

File: README.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+- Add a breakpoint within the file that you would like to watch and deploy a capability
+  ```bash
+  kubectl apply -f deploy/example-capability.yml
+  ```
+- Enjoy !  
+
+## Release
```
Comment (metacosm) [inline README.md:89]: This section should simply point to: https://github.com/halkyonio/operator-framework/tree/master/plugins#using-plugins-in-halkyon

File: README.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
+# Instructions to create a new plugin
```
Comment (metacosm) [inline README.md:1]: This is missing a link to https://github.com/halkyonio/operator-framework/tree/master/plugins

**Review by metacosm — CHANGES_REQUESTED** [review]
Please use semantic commits. Commit type should be lowercased…

File: README.md (hunk 1 above)
Comment (cmoulliard) [reply README.md:78]: I was not able to debug without them. Can you make a test again ? @metacosm

File: README.md (hunk 2 above)
Comment (cmoulliard) [reply README.md]: Fixed

File: README.md (hunk 3 above)
Comment (cmoulliard) [reply README.md]: Fixed

File: README.md (hunk 5 above)
Comment (cmoulliard) [reply README.md:89]: Fixed

File: README.md (hunk 6 above)
Comment (cmoulliard) [reply README.md:1]: Fixed

**Comment by cmoulliard** [comment]
Remark: As we will refactor the documentation and probably moves this project under `operator-framework`, then it will be needed to change the link of this README.md file !

File: README.md (hunk 1 above)
Comment (metacosm) [reply README.md:78]: I've been debugging the kubedb plugin without these flags without issue.

**Review by metacosm — APPROVED** [review]

**Pull request merged**